### PR TITLE
Initial repo structure based on GSA GRACE

### PR DESCRIPTION
**Pull request by JJediny**
While this is alot in one commit it is based on GSA GRACE's latest successfully build and defines a starting point for work. All PR/commits moving forward will be small/incremental.

### CODE REVIEW

File: terraform/cloudcheckr/cloudcheckr.tf [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+      "swf:ListWorkflowTypes", 
+      "workspaces:DescribeWorkspaceBundles", 
+      "workspaces:DescribeWorkspaceDirectories", 
+      "workspaces:DescribeWorkspaces"
+    ],
+    "Resource":"*"
+  }]
+}
+EOF
+}
```
Comment (afeld) [inline terraform/cloudcheckr/cloudcheckr.tf]: Not a blocker: to keep the files small and have syntax highlighting, I recommend putting policies in an external JSON file and [reading it in to the resource](https://www.terraform.io/docs/configuration/functions/file.html).

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - run:
+          name: Install make
+          command: apk add --update make
+      - run:
+          name: Install python3
+          command: apk add --update python3
```
[the remark this answers is not in the document]
Comment (JJediny) [reply .circleci/config.yml]: combined to one command

File: terraform/Makefile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	@exit 1
+else
+	@echo "appenv: $(TF_VAR_appenv)"
+endif
+ifeq ($(strip $(backend_key)),)
+	@echo "backend_key must be provided"
+	@exit 1
+endif
+
+apply: plan
```
[the remark this answers is not in the document]
Comment (JJediny) [reply terraform/Makefile]: done

File: terraform/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# TTS CORE AWS Cross Account IAM
```
[the remark this answers is not in the document]
Comment (JJediny) [reply terraform/README.md]: moved to main README

File: terraform/app/output.tf [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  value = "${var.appenv == "staging" ? 1 : 0}"
+}
+
+output "is_test" {
+  value = "${var.appenv == "test" ? 1 : 0}"
+}
+
+output "is_development" {
+  value = "${var.appenv == "development" ? 1 : 0}"
+}
```
[the remark this answers is not in the document]
Comment (JJediny) [reply terraform/app/output.tf]: relic of GRACE var names didnt want to change for consistency

File: terraform/app/README.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# App
```
[the remark this answers is not in the document]
Comment (JJediny) [reply terraform/app/README.md]: leaving for now to placehold the discussion of level of IAM per environment

File: terraform/guardduty/guardduty.tf [hunk 7]
```diff
@@ -0,0 +1,3 @@
+resource "aws_guardduty_detector" "master-tts" {
+  enable = true
```
[the remark this answers is not in the document]
Comment (JJediny) [reply terraform/guardduty/guardduty.tf]: changed this to a variable with default off/false

File: terraform/cloudcheckr/cloudcheckr.tf (hunk 1 above)
Comment (JJediny) [reply terraform/cloudcheckr/cloudcheckr.tf]: keeping for now at it builds a single json policy based on declared variables vs pulls in a json blob

**Comment by JJediny** [comment]
merging for now as this is only a baseline starting point

**Pull request merged**